metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -341,22 +359,25 @@ public void replay(RemoveTopicRecord record) {
         // Delete the configurations associated with this topic.
         configurationControl.deleteTopicConfigs(topic.name);
 
-        // Remove the entries for this topic in brokersToIsrs.
-        for (PartitionRegistration partition : topic.parts.values()) {
+        for (Map.Entry<Integer, PartitionRegistration> entry : topic.parts.entrySet()) {
+            int partitionId = entry.getKey();
+            PartitionRegistration partition = entry.getValue();
+
+            // Remove the entries for this topic in brokersToIsrs.
             for (int i = 0; i < partition.isr.length; i++) {
                 brokersToIsrs.removeTopicEntryForBroker(topic.id, partition.isr[i]);
             }
-            if (partition.leader != partition.preferredReplica()) {
-                preferredReplicaImbalanceCount.decrement();
-            }
+
+            imbalancedPartitions.remove(new TopicIdPartition(record.topicId(), partitionId));
+
```

Comment:
nit: whitespace

Reply:
What do you mean?

Reply:
Just that you've got blank lines before and L371, not really a problem just not sure if that's what you intended